jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -5435,17 +5615,21 @@ public boolean isNestmateOf(Class<?> that) {
 /*[IF JAVA_SPEC_VERSION < 15]
  * @throws LinkageError if there is any problem loading or validating a nest member or the nest host
 /*[ENDIF]
+/*[IF JAVA_SPEC_VERSION < 24]
  * @throws SecurityException if a returned class is not the current class, a security manager is enabled,
  *	the caller's class loader is not the same or an ancestor of that returned class, and the
  * 	checkPackageAccess() denies access
+/*[ENDIF] JAVA_SPEC_VERSION < 24
  * @return the host class of the receiver.
  */
 @CallerSensitive
-public Class<?>[] getNestMembers() throws
+public Class<?>[] getNestMembers()
 /*[IF JAVA_SPEC_VERSION < 15]*/
-LinkageError,
+		throws LinkageError, SecurityException
+/*[ELSEIF JAVA_SPEC_VERSION < 24] */
+		throws SecurityException
 /*[ENDIF] JAVA_SPEC_VERSION < 15 */
-SecurityException {
+{
```

Comment:
Doesn't the call to securityManager.checkPackageAccess() below need to be removed?